.github/workflows/update_goldens.yml

```diff
@@ -0,0 +1,42 @@
+name: Update Goldens
+on: 
+  workflow_dispatch:
+    inputs:
+      commit:
+        type: boolean
+        default: true
+        required: true
+        description: Commit the new Goldens to this branch.
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    permissions:
+      contents: write
+    steps:
+      - name: 📚 Git Checkout
+        uses: actions/checkout@v3
+
+      - name: 🏗 Build and test Sweyer
+        uses: ./.github/workflows/build_and_test
+        with:
+          flutter_channel: stable
+          flutter_version: 2.10.5
+          testing_arguments: --update-goldens
+
+      - name: 📁 Save updated Golden test artifacts
+        uses: actions/upload-artifact@v3
+        with:
+          name: golden-test-updated
+          path: test/golden/goldens
+
+      - name: 💂 Verify that ${{ github.actor }} has permission to commit
+        if: ${{ github.event.inputs.commit == 'true' }}
+        shell: bash
+        run: git log --author='^${{ github.actor }} <\S*>$' --max-count=1 --exit-code > /dev/null && { echo '⚠ You have no commits on this branch and therefore no permission to commit ⚠' ; exit 1; } || exit 0
```

Comment:
This doesn't verify actual permissions - for example you don't have permissions to commit into any of the branches, but you have commits with your signature in the tree.

I would honestly prefer that we rather just limit this workflow to repository admins, so I as reviewer can trigger it on a pull request before I merge, after I reviewed all the goldens manually. It will regenerate all goldens and it's not ideal but I think it's an okay policy and would avoid potential flakes that could happen because of tolerance gap.

Reply:
By admins I meant collaborators, actually

Reply:
> This doesn't verify actual permissions

Yes, I mentioned this in the PR description briefly.This was the easiest way that I could think of that allows people to commit to their own branches or "collaborators" which have previously committed. I was hesitant to use the [actual collaborators list from the GitHub repository](https://github.com/nt4f04uNd/sweyer/graphs/contributors), because it is empty, since you squash all commits on a merge request, so GitHub counts the code as your contribution.

I will look into an alternative solution, because it would be nice if someone at least could commit into their own branches.

Reply:
> I was hesitant to use the [actual collaborators list from the GitHub repository](https://github.com/nt4f04uNd/sweyer/graphs/contributors), because it is empty, since you squash all commits on a merge request, so GitHub counts the code as your contribution.

Github doesn't count squashed commits as my contribution, since they are signed with the email of the pull request author, which is you. I might be co-author of the squshed commit, but that is only if I commited to pull request branch.

This list also doesn't display who is a collaborator on the repository, this is a list of all contributors.
Also, Github counts contibutions only on master branches of non-fork repositories, hence it doesn't display you in the list, until https://github.com/nt4f04uNd/sweyer/commits/1.0.8 is merged into master.

Reply:
Ok, thanks for the clarification, I just assumed that the list would be empty based on what I could see. I would still like to try to allow people to commit to their own branches, its just very convenient. How about this:

We can check if someone is a contributor using [`actions-cool
/check-user-permission`](https://github.com/actions-cool/check-user-permission)
If the branch has a parent, we allow everyone who has made a commit on that branch up to it's parent, meaning we are looking just at the new commits on the branch.

Would that be ok with you?

Reply:
Actually, I wonder if checking for the write user-permission is enough...

Reply:
@nt4f04uNd could you please test if you can run the workflow with auto commit against `master` in [my fork](https://github.com/Abestanis/sweyer/actions/workflows/update_goldens.yml)? Also, try to open a pull request and check the auto commit against that branch.

Reply:
Hang on, it's not working just yet.

Reply:
Ok, you can try now

Reply:
> could you please test if you can run the workflow with auto commit against master in [my fork](https://github.com/Abestanis/sweyer/actions/workflows/update_goldens.yml)?

I cannot, there is no button to run a workflow for me.

I see that you try to disable only the step that actually commits goldens, but I would prefer to make this entire workflow only accessible to collaborators, it otherwise incentivizes to run it for no reason - reviewers should decide if new goldens are good or not.

Reply:
Ah, nevermind, workflow_dispatch is by default only accessible to collaborators